tools/kliopt/org/jetbrains/kliopt/KliOpt.kt

```diff
@@ -17,6 +17,21 @@
 
 package org.jetbrains.kliopt
 
+enum class DeprecatedState {
+    ALL {
+        override fun helpMessage(descriptor: Descriptor): String = "Warning: Option/argument -${descriptor.longName} is deprecated!"
+    },
+    ONLY_SHORT {
+        override fun helpMessage(descriptor: Descriptor): String =
+                "Warning: Short form -${(descriptor as? OptionDescriptor)?.shortName} of option -${descriptor.longName} is deprecated!"
```

Comment:
I believe that it would be more user-friendly to give a hint here, e.g. `"Warning: -h option is deprecated, use -header instead"`.